packages/dds/sequence/src/sharedString.ts

```diff
@@ -223,6 +224,7 @@ export class SharedString
 	 * @param combiningOp - Optional. Specifies how to combine values for the property, such as "incr" for increment.
 	 */
 	public annotateMarker(marker: Marker, props: PropertySet, combiningOp?: ICombiningOp) {
+		assert(props.markerId === undefined, "Cannot change the markerId of an existing marker");
```

Comment:
I think you need to check that the key markerId does not exist at all. Setting the value of the key to undefined will cause PropertyManager to clear the existing property.

Reply:
Add tests for annotating with markerId=null and undefined too.